pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,35 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, ingressCertificate *corev1.LocalObjectReference, clusterDomain string) error {
+	isManagedDomain := dns.IsManagedDomain(clusterDomain)
+
+	if ingressCertificate == nil {
+		if isManagedDomain {
+			return errors.New("missing ingress certificate for cluster with managed domain")
```

Comment:
would be slightly better to extract it into a var at package level so you can reuse it in the test

Reply:
In fact, this is something I asked Mikalai some months ago: "why do we compare the error string in test cases instead of declaring the errors at package level and check that the error is the same". We could use == or errors.Is in test functions instead of hardcoded strings.

So I agree about putting the error at the package level. 
Doing it now.